Objects/codeobject.c

```diff
@@ -220,9 +222,56 @@ PyCode_New(int argcount, int kwonlyargcount,
     co->co_zombieframe = NULL;
     co->co_weakreflist = NULL;
     co->co_extra = NULL;
+
+    co->co_opt_flag = 0;
+    co->co_opt_size = 0;
+    co->co_opt_opcodemap = NULL;
+    co->co_opt = NULL;
     return co;
 }
 
+int
+_PyCode_InitOptCache(PyCodeObject *co)
+{
+    Py_ssize_t co_size = PyBytes_Size(co->co_code) / sizeof(_Py_CODEUNIT);
+    co->co_opt_opcodemap = (unsigned char *)PyMem_Calloc(co_size, 1);
+    if (co->co_opt_opcodemap == NULL) {
+        return -1;
+    }
+
+    _Py_CODEUNIT *opcodes = (_Py_CODEUNIT*)PyBytes_AS_STRING(co->co_code);
+    Py_ssize_t opts = 0;
+
+    for (Py_ssize_t i = 0; i < co_size;) {
+        unsigned char opcode = _Py_OPCODE(opcodes[i]);
+        i++;  // 'i' is now aligned to ceval/INSTR_OFFSET()
+
+        // TODO: LOAD_METHOD, LOAD_ATTR
+        if (opcode == LOAD_GLOBAL) {
+            co->co_opt_opcodemap[i] = ++opts;
+            if (opts > 254) {
```

Comment:
Perhaps use 16-bit indices instead?

Reply:
To maximize "performance benefit / memory usage", I prefer using 1 byte for each instructions, instead of two bytes.